tests/integration/adapter-identifier-tests.js

```diff
@@ -0,0 +1,307 @@
+          method: 'POST',
+          body: '{}',
+          url: 'http://a:b@localhost:4000/pathname?query=param'
+        }
+      },
+      xhr: {
+        id: '683833ae6f08ec5c77a7cd8620cd35c9',
+        identifiers: {
+          headers: { 'content-type': 'application/json;charset=utf-8' },
+          method: 'POST',
+          body: '{}',
+          url: 'http://a:b@localhost:4000/pathname?query=param'
+        }
+      }
+    });
+
+    describe('url', function() {
+      testConfiguration('url.protocol', false, {
+        fetch: {
+          id: '162f8e80be1b290e52812bfebf528f6c',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: '//a:b@localhost:4000/pathname?query=param'
+          }
+        },
+        xhr: {
+          id: '162f8e80be1b290e52812bfebf528f6c',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: '//a:b@localhost:4000/pathname?query=param'
+          }
+        }
+      });
+
+      testConfiguration('url.hostname', false, {
+        fetch: {
+          id: 'e4b32d97c51e32d71de1294a82f6b04b',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@:4000/pathname?query=param'
+          }
+        },
+        xhr: {
+          id: 'e4b32d97c51e32d71de1294a82f6b04b',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@:4000/pathname?query=param'
+          }
+        }
+      });
+
+      testConfiguration('url.pathname', false, {
+        fetch: {
+          id: '12d416e92eff0e8543ede886919c02c2',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000?query=param'
+          }
+        },
+        xhr: {
+          id: '12d416e92eff0e8543ede886919c02c2',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000?query=param'
+          }
+        }
+      });
+
+      testConfiguration('url.port', false, {
+        fetch: {
+          id: '147ca2615501f226981619c32341699b',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost/pathname?query=param'
+          }
+        },
+        xhr: {
+          id: '147ca2615501f226981619c32341699b',
+          identifiers: {
```

Comment:
Decided to assert also against the identifiers, will help when we need to triage why an ID changed.